src/server/test/routes/unitAddParamsTest.js

```diff
@@ -0,0 +1,157 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//
+const { expect } = require('chai');
+// SHL: recreateDB does not seem to be used.
+// BM: recreateDB deleted
+const { chai, mocha, app, testDB } = require('../common');
+const Unit = require('../../models/Unit');
+// SHL: This does not seem to be used.
+// BM: THis is for the first way to test the params
+//This is the first way to test endpoint
+const { validateString, validateInt, validateBool, validateMinMaxRelation } = require('../util/vaidationHelpers');
+
+//This is the end point we use to test in this file.
+const ADD_UNIT = '/api/units/addUnit';
+
+const basePayload = {
+	name: 'Valid Name',
+	identifier: 'valid_id',
+	unitRepresent: 'flow',
+	secInRate: 60,
+	typeOfUnit: 'unit',
+	suffix: 'L/s',
+	displayable: 'all',
+	preferredDisplay: true,
+	note: 'Note text',
+	minVal: 0,
+	maxVal: 100,
+	disableChecks: 'reject_bad'
+};
+
+// SHL: I'm curious why these tests are so different than the edit verify and seem to cover different tests.
+// BM: In this 2nd way, I have the validation function series to auto apply some incorrect params and test it.
+mocha.describe('Validation - /addUnit', () => {
+	mocha.it('should validate string fields', async () => {
+		// Based on schema: name VARCHAR(50), NOT NULL
+		await validateString({
+			field: 'name',
+			endpoint: ADD_UNIT,
+			basePayload,
+// SHL: name also has a min of 1 similar to identifier.
+// BM: changed
+			maxLength: 50,
```

Comment:
This one has max first but others have it after min. Small point but nice if this was the same with max after min.